ddl/index.go

```diff
@@ -975,7 +977,60 @@ func (w *worker) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgI
 	}
 	defer closeAddIndexWorkers(idxWorkers)
 
-	return w.buildKVRangesIndex(t, idxWorkers, job, reorgInfo)
+	finish := false
+	for !finish {
+		err := w.buildIndexForReorgInfo(t, idxWorkers, job, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		finish, err = w.updateReorgInfo(t, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// updateReorgInfo will find the next partition according to current reorgInfo.
```

Comment:
Add a comment for the first return value.

Reply:
Done